src/packet/BinaryWriter.js

```diff
@@ -1,137 +1,62 @@
-﻿'use strict';
+'use strict';
 /*
  * Simple BinaryWriter is a minimal tool to write binary stream with unpredictable size.
  * Useful for binary serialization.
  *
- * Copyright (c) 2016 Barbosik
- * 
- * Licensed under the Apache License, Version 2.0 (the "License");
- * you may not use this file except in compliance with the License.
- * You may obtain a copy of the License at
- * 
- *     http://www.apache.org/licenses/LICENSE-2.0
- * 
- * Unless required by applicable law or agreed to in writing, software
- * distributed under the License is distributed on an "AS IS" BASIS,
- * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
- * See the License for the specific language governing permissions and
- * limitations under the License.
- * 
- * 
+ * Copyright (c) 2016 Barbosik https://github.com/Barbosik
+ * License: Apache License, Version 2.0
  */
 
-function BinaryWriter(size) {
-    if (!size || size <= 0) {
-        size = Buffer.poolSize / 2;
-    }
-    this._buffer = new Buffer(size);
+function BinaryWriter() {
+    this._buffer = parseInt(process.version[1]) < 6 ? new Buffer(1e5) : Buffer.allocUnsafe(1e5);

```

Comment:
"Incredibly" wrong?
Wrong with 3 points out of 8 is "incredibly" wrong?

> All I need to do to crash, is have many bots join with a large name and skin. Then the buffer would be overloaded at one point, and there you go. 
- Great point. Oh hey, solvable with two lines of code.

>Also, 100000 bytes is not a good size. Do you really want each player to have 100000 bytes? Thats aweful.
- Please elaborate the meaning of *Do you really want each player to have 100000 bytes?*

> Also, allocation checks shouldnt be expensive, unless barbosik made them wrong.
- Barbosik's allocation check created a new `Buffer` every time a byte was added. He then inserted the new bytes into that buffer and merged the buffer with the entire buffer for the packet. These operations can and do add up.

> Also, precalculating is easier than you think. Mine doesnt loop again (opnagar).
- In some cases yes, it doesn't. Another good point. Bonus kudos for mentioning your "semi-private" project.

> And for your last comment, no, it is not fast enough, for everyone; look at all the issues with lag.
- The reason is bandwidth limits.
- Many of them host it out of cloud9 or codeanywhere. They're environments. They aren't for server hosting. The servers are hosted under virtual machines. Every virtual machine has bandwidth limits. And you think those servers need to support 100 players? I was one of them for a few days and they rightfully terminated my account on c9.